sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -133,14 +133,32 @@ class SparkSession private(
   /** Tag to mark all jobs owned by this session. */
   private[sql] lazy val sessionJobTag = s"spark-session-$sessionUUID"
 
+  /**
+   * A UUID that is unique on the thread level. Used by managedJobTags to make sure that the same
+   * use tag do not overlap in the underlying SparkContext/SQLExecution.
+   */
+  private[sql] lazy val threadUuid = new ThreadLocal[String] {
```

Comment:
Hm, would have to be InheritableThreadLocal. I believe that's how Spark Connect works.

Reply:
Thanks, fixed!
I was confused about thread vs session. The `Inheritable` here is for child threads.